evalml/tests/component_tests/test_components.py

```diff
@@ -1631,13 +1628,13 @@ def test_component_modifies_feature_or_target():
     for component_class in all_components():
         if (
             issubclass(component_class, BaseSampler)
-            or issubclass(component_class, TargetTransformer)
+            or hasattr(component_class, "inverse_transform")
```

Comment:
Open question: Do you think this will hold true long term? It's true for our components now but... 👀

Reply:
Hmm, I do think that if we have an `inverse_transform` method, then it'll modify the target rather than the features. It doesn't seem to make sense to have a transformer have an 'inverse_tranform` method for the features. I also don't think it'd make sense to introduce an `inverse_transform` method if we don't alter the target at all. Logically, it seems like this should still hold long term.

Reply:
Should we add `inverse_transform` as a method to `Transformer`. The "default" behavior is a no-op and if we want to enforce that transformers that modify the target override that behavior we can do something like

```python
def inverse_transform(self, y):
    if self.modifies_target:
        raise NotImplementedError("Transformers that modify the target must override inverse_transform")
    return y
```

Or we don't make this change? Maybe I was being crazy when I filed the issue lol

Reply:
I think Freddy's suggestion is a good one!